common/JackRequest.h

```diff
@@ -1310,16 +1355,18 @@ struct JackSessionNotifyRequest : public JackRequest
     jack_session_event_type_t fEventType;
     int fRefNum;
 
-    JackSessionNotifyRequest()
+    JackSessionNotifyRequest() : fEventType(JackSessionSave), fRefNum(0)
     {}
     JackSessionNotifyRequest(int refnum, const char* path, jack_session_event_type_t type, const char* dst)
             : JackRequest(JackRequest::kSessionNotify), fEventType(type), fRefNum(refnum)
     {
+        memset(fPath, 0, sizeof(fPath));
+        memset(fDst, 0, sizeof(fDst));
         snprintf(fPath, sizeof(fPath), "%s", path);
+        fPath[JACK_MESSAGE_SIZE] = 0;
         if (dst) {
             snprintf(fDst, sizeof(fDst), "%s", dst);
-        } else {
-            fDst[0] = '\0';
+            fDst[JACK_CLIENT_NAME_SIZE] = 0;
```

Comment:
Yes, this is semantically correct.
- I dropped the old _} else {_ condition at line 1321,
- and the old instruction _fDst[0] = '\0';_ at line 1322 is now performed by the new instruction _memset(fDst, 0, sizeof(fDst));_ at line 1364.

Both instructions:
1. _fPath[JACK_MESSAGE_SIZE] = 0;_
2. _fDst[JACK_CLIENT_NAME_SIZE] = 0;_
make sure strings are NUL terminated. In fact, those paranoid instructions can be dropped.